### PR TITLE
Add a YOLOv3 analyser

### LINKED ISSUE #15: add analyser for yolov3
Yolov3 inference needs to be made available, to apply on images.

### PR DESCRIPTION
**Pull request by breezykermo**
closes #15 

This adds an analyser to run object detection inference using [YOLOv3](https://pjreddie.com/media/files/papers/YOLOv3.pdf). The code is ported from [this PyTorch implementation](https://github.com/eriklindernoren/PyTorch-YOLOv3).

A couple of questions/concerns about this PR that need review:
* The original code is availlable under [GPL](https://en.wikipedia.org/wiki/GNU_General_Public_License). mtriage is under the DoNoHarm license, which is a derivation of the GPL I'm pretty sure. What do we need to do to make sure @eriklindernoren's code is appropriately referenced? 
* This is the first analyser that introduces GPU use through CUDA. This is relatively painless thanks to nvidia-docker, but it does add some complexity in the build process. Essentially, I've added the ability to add a `--gpu` flag for both `build` and `develop` commands, which causes mtriage to build from a PyTorch/CUDA base rather than straight up ubuntu. All of this stuff is necessary, but if there are any ideas regarding how it could be managed more elegantly, I'm all ears. The YoloV3 analyser should still work without a GPU, although admittedly I haven't tested this yet.
* The actual YOLOv3 code can probably be tidied further, as I haven't really scrutinised it at all beyond getting it working. I'll take at least one extra pass tomorrow, and likely more if I can see other TODOs.

There are also a couple of TODOs before this can be merged:
- [x] write JSON object to output with appropriate fields for mtriage-view.
- [x] ~pass in hyperparameters as analyser config options~
- [x] ~add a `visualise` flag that, when true, returns the functionality that creates images with bounding box and label around predicted objects. This creation takes a significant amount of time, and isn't necessary for use with mtriage-view.~
- [x] update the CI scripts to push `forensicarchitecture/mtriage:dev-gpu` on merge into master along with `forensicarchitecture/mtriage:dev`.
- [x] ~add tests to reflect the new optionality in the outer run.py with `--gpu`.~

### CODE REVIEW

**Comment by breezykermo** [comment]
@samludford this is ready for review. Due to the fact that a lot of the outer run.py is currently untested, I have made a separate issue for the last crossed-out point in the checklist #62 which will be in this release.

The first two crossed-out points in the checklist are inessential for us at the moment, but would be useful for making yolov3 more general purpose. We can potentially list them as "help wanted" issues for community, what do you think?

**Comment by breezykermo** [comment]
One thing to note is that this PR almost doubles the build time in CI, as it confirms that both CPU and GPU docker images build successfully on each run. We perhaps need to start thinking more about mtriage builds with only certain selectors/analysers enabled...

**Comment by breezykermo** [comment]
GPU/CPU build stuff will be improved in #63.

**Comment by breezykermo** [comment]
Going to close this and PR #63 with its commits, as there's a load fixed up there.
